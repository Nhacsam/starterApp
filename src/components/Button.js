// @flow

import React, { PureComponent } from 'react';
import { Text, StyleSheet, ActivityIndicator } from 'react-native';
import Touchable from './Touchable';

import theme from 'starterApp/src/theme';

class Button extends PureComponent {
  static defaultProps: PropsTypes = {
    onPress: () => {},
    fetching: false,
    borderless: false,
  };

  props: PropsTypes;

  render() {
    const { style, children, text, textStyle, fetching, ...rest } = this.props;
    let content;
    if (fetching) {
      content = <ActivityIndicator color={theme.colors.overPrimary} />;
    } else if (text) {
      content = <Text style={[styles.text, textStyle]}>{text.toUpperCase()}</Text>;
    } else {
      content = children;
    }

    return (
      <Touchable style={[styles.button, style]} {...rest}>
        {content}
      </Touchable>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    height: 40,
    backgroundColor: theme.colors.primary,
    paddingHorizontal: 8,
    borderRadius: 5,
    marginVertical: 8,
  },
  text: {
    ...theme.fonts.button,
    color: theme.colors.overPrimary,
    textAlign: 'center',
  },
});

type PropsTypes = {
  text?: string,
  onPress: () => void,
  children?: React.Element<*>,
  style?: StyleSheet.Styles | Array<StyleSheet.Styles>,
  textStyle?: StyleSheet.Styles | Array<StyleSheet.Styles>,
  fetching: boolean,
  borderless: boolean,
};

export default Button;
