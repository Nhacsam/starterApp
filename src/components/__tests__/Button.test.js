import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';
import Components from 'components';

describe('Button', () => {
  it('should render the button', () => {
    const tree = renderer.create(<Button text="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should export a Button in components', () => {
    expect(Components.Button).toBe(Button);
  });
});
