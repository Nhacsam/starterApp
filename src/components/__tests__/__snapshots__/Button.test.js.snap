// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginVertical": 8,
      "minHeight": 48,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#05A5D1",
        "borderRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 13,
            "fontWeight": "700",
            "textAlign": "center",
          },
        ]
      }
    >
      HELLO
    </Text>
  </View>
</View>
`;
